### ISSUE TITLE
Conv2d with single input channel incorrect (on Apple M1 chip)

### ISSUE THREAD

**Issue by jack-willturner** [pytorch/pytorch#60535]
## 🐛 Bug

When the input has a single channel, `conv2d` seems to only produce the first output channel correctly, and the rest are zeros everywhere.

## To Reproduce

Code to reproduce the behavior:

```python
import torch
import torch.nn as nn

input_example = torch.zeros(1,1,3,3)
# set the middle pixel to be a one (the rest are zeros)
input_example[:,:,1,1] = 1.

conv = nn.Conv2d(1, 5, kernel_size=3, bias=False)
# set the conv weights to be five 3x3 identity matrices

conv.weight.data = torch.eye(3,3).unsqueeze(0).repeat(5,1,1,1)
print(conv(input_example))
```

Produces the output:

```
tensor([[[[ 1.0000e+00]],
         [[-1.0842e-19]],
         [[ 0.0000e+00]],
         [[-1.0842e-19]],
         [[ 1.1210e-44]]]])
```

## Expected behavior

I would expect each channel in the above to produce a 1 (instead the first channel is a 1 and the rest are ~0s).

Increasing the number of input channels makes the problem go away, so I suspect it is linked to #56145. 

## Environment

```
PyTorch version: 1.8.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 11.3.1 (arm64)
GCC version: Could not collect
Clang version: 11.1.0
CMake version: version 3.20.2
Libc version: N/A

Python version: 3.8 (64-bit runtime)
Python platform: macOS-11.3.1-arm64-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.3
[pip3] torch==1.8.0
[pip3] torchvision==0.2.2
[conda] numpy                     1.20.3           py38hbf7bb01_1    conda-forge
[conda] pytorch                   1.8.0           cpu_py38hc43b888_1    conda-forge
[conda] torchvision               0.2.2                      py_3    pytorch
```

cc @malfet @albanD

**Comment by VitalyFedyunin** [comment]
Need to discuss new label for Apple M1

**Comment by malfet** [comment]
@VitalyFedyunin I've been using combination of MacOS + ARM
And this sounds like a duplicate of https://github.com/pytorch/pytorch/issues/60176

**Comment by malfet** [comment]
@jack-willturner is there a reason why you need to use PyTorch-1.8? I could not reproduce it with top-of-tree build nor using PyTorch-1.9 binaries
